[JENKINS-18884] Remove 'People' view

daniel-beck opened the pull request:
See [JENKINS-18884](https://issues.jenkins.io/browse/JENKINS-18884) and https://github.com/jenkinsci/jenkins/pull/9052#issuecomment-2007898267 for previous context. This is the core change corresponding to the creation of a new plugin as suggested in the linked comment.

Plugin sketch: https://github.com/daniel-beck/people-view-plugin

### Usage check

A quick GH search indicates just two affected plugins:

* https://plugins.jenkins.io/graphql-server/ (recently [disowned](https://groups.google.com/g/jenkinsci-dev/c/IA4THcgsxc0/m/7sCmVqDZAgAJ))
* https://plugins.jenkins.io/jabber-server-plugin/ (last released in 2014, unresolved vulnerability).

usage-in-plugins only finds `jabber-server-plugin`.

<details>

As `-M` and `-C`:

```
hudson/model/View$People
hudson/model/View$AsynchPeople
hudson/model/View$AsynchPeople$People
hudson.model.View#getAsynchPeople
hudson.model.View#getPeople
hudson.model.View#hasPeople
jenkins.model.Jenkins#getAsynchPeople
jenkins.model.Jenkins#getPeople
jenkins.model.Jenkins#hasPeople
```

#### Result

[jabber-server-plugin:1.9](https://plugins.jenkins.io/jabber-server-plugin)
Classes
[hudson/model/View$People](http://javadoc.jenkins.io/hudson/model/View.People.html)
Methods
[jenkins/model/Jenkins#getPeople()Lhudson/model/View$People;](http://javadoc.jenkins.io/jenkins/model/Jenkins.html#getPeople%28%29)

</details>

### Testing done

Navigated to `/people/` and `/asynchPeople/`, saw the new placeholder page.

Ran https://github.com/daniel-beck/people-view-plugin/ as of https://github.com/daniel-beck/people-view-plugin/commit/3bd4f1ddd531ddeac30b71a8c44ffd0983e7bb0a and confirmed I'm seeing the plugin-contributed views.

### Proposed changelog entries

- Remove "People" view. Administrators can install the new People View plugin to restore this functionality.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/core-pr-reviewers.
-->

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

daniel-beck committed:
[JENKINS-18884] Remove 'People' view

daniel-beck commented on core/src/main/resources/hudson/model/View/sidepanel.jelly:
```diff
@@ -43,14 +43,6 @@ THE SOFTWARE.
       </st:include>
 
-      <j:choose>
-        <j:when test="${it.class.name=='hudson.model.AllView'}">
-          <l:task href="${rootURL}/asynchPeople/" icon="icon-user icon-md" title="${%People}"/>
-        </j:when>
```
FTR this is a bug IMO, as AllView can exist in folders. Note how https://weekly.ci.jenkins.io/job/folder/ links to `/asynchPeople/`.

daniel-beck commented on core/src/main/resources/hudson/model/User/sidepanel.jelly:
```diff
@@ -31,7 +31,6 @@ THE SOFTWARE.
   <l:header />
   <l:side-panel>
     <l:tasks>
-      <l:task contextMenu="false" href="${rootURL}/asynchPeople/" icon="symbol-people" title="${%People}"/>
```
This is an awkward case as the people view basically represents the list of users, of which an individual user is an entry. Unless using the built-in Jenkins' own user database, admins no longer have a users list with this change. We should probably add a `ManagementLink` before long to iterate over (actual) users.

jglick replied:
> Unless using the built-in Jenkins' own user database, admins no longer have a users list with this change.

They do not have such a list currently. `AsynchPeople` lists SCM users which are not in the security realm; and does not list users which _are_ in the security realm but have no user record on disk. It is simply useless for this purpose.

daniel-beck replied:
> but have no user record on disk

Right, but anyone who's logged in once has such a record, which makes it a viable (if crappy and slow) entrypoint to getting to someone that's not directly entering the URL.

daniel-beck committed:
Add placeholder pages when someone tries to access removed URLs

daniel-beck committed:
Remove obsolete UserInfo class, inapplicable tests

jglick commented on core/src/main/java/jenkins/model/PeopleRedirect.java:
```diff
@@ -0,0 +1,78 @@
+        rsp.forward(this, "", req);
+    }
+
+    @Override
+    public String getIconFileName() {
+        return null;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return null;
+    }
```
This hides the links from the sidebar. If that is what you meant to do, you do not need this `Action` or TAF at all. Simply create `View/people.jelly` and `View/asynchPeople.jelly`. (not sure about trailing `/`)

daniel-beck replied:
>  (not sure about trailing `/`)

That's the problem I encountered, `/people/` gets 404 with `people.jelly`. Since that's the current (e.g. bookmarked) URL, I went with this instead.

jglick commented on core/src/main/java/jenkins/model/PeopleRedirect.java:
```diff
@@ -0,0 +1,78 @@
+public abstract class PeopleRedirect implements Action {
+    /* Ensure all requests end in the index view */
+    public void doDynamic(StaplerRequest req, StaplerResponse rsp) throws ServletException, IOException {
+        // TODO this has broken breadcrumbs -- also, do we even need this?
```
No, `index.jelly` would be served automatically, no web method required. Unless I am missing something about what you are trying to accomplish with this action. A Javadoc comment would be helpful.

daniel-beck replied:
Sending a reasonable response to `/*people/api/` and further URLs with that prefix.

jglick commented on test/src/test/java/jenkins/security/stapler/GetterMethodFilterTest.java:
```diff
@@ -266,19 +265,19 @@ public void testWithReturnMultiple_map() throws Exception {
-    public void testWithReturnCoreObject_people() throws Exception {
-        assertReachableWithoutOk("testWithReturnCoreObject/people/");
-    }
+//
+//    @TestExtension
+//    public static class TestWithReturnCoreObject extends AbstractUnprotectedRootAction {
+//        public View.People getPeople() {
+//            // provide an index jelly view
+//            return new View.People(Jenkins.get());
+//        }
+//    }
+//
+//    @Test
+//    public void testWithReturnCoreObject_people() throws Exception {
+//        assertReachableWithoutOk("testWithReturnCoreObject/people/");
+//    }
```
just
```suggestion
```

daniel-beck replied:
Just a reminder to look for an alternative to retain test coverage, see the section "Work in progress" in the PR comment.

daniel-beck committed:
Restore test with a different core-contributed object

It's not entirely clear what this test signifies, given the
existence of `testTopLevelItemIsLegal`, but whatever.

daniel-beck committed:
Remove probably unnecessary code

jglick commented on core/src/main/resources/jenkins/model/PeopleRedirect/index.properties:
```diff
@@ -0,0 +1 @@
+blurb = The "People" view has been moved into the <a href="https://plugins.jenkins.io/people-view" target="_blank" rel="noopener noreferrer">People View</a> plugin.
```
```suggestion
blurb = The "People" view has been moved into the <a href="https://plugins.jenkins.io/people-view/" target="_blank" rel="noopener noreferrer">People View</a> plugin.
```

MarkEWaite commented:
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

/label ready-for-merge

timja commented:
Likely caused https://issues.jenkins.io/browse/JENKINS-73185

also reported on https://community.jenkins.io/t/people-view-plugin-very-slow/14898

daniel-beck commented:
> caused

Disagree. The same problem would be present if you install the People View plugin on Jenkins releases where this was still a core feature.

timja commented:
> Disagree. The same problem would be present if you install the People View plugin on Jenkins releases where this was still a core feature.

That plugin only has a use due to this pull request and part of the justification for acceptance was a 'detached' plugin. If the plugin is fundamentally broken at some scale then this pull request has broken this functionality for users, no?

> This plugin provides the "People" view removed from Jenkins core

daniel-beck commented:
I disagree with this inflationary use of "regression" or "causes" I'm seeing recently. This PR specifically does _not_ cause JENKINS-73185 because it does not install the plugin.
